pkg/render/guardian.go

```diff
@@ -312,6 +322,45 @@ func (c *GuardianComponent) clusterRoleBinding() *rbacv1.ClusterRoleBinding {
 	}
 }
 
+func (c *GuardianComponent) Role() *rbacv1.Role {
```

Comment:
```suggestion
func (c *GuardianComponent) secretsRole() *rbacv1.Role {
```

Doesn't need to be public? And a mildly more descriptive name.